collector/mdadm_linux.go

```diff
@@ -138,12 +163,20 @@ func parseMdstat(mdStatusFilePath string) ([]mdStatus, error) {
 		}
 		currentMD = mainLine[0]               // name of md-device
 		isActive := (mainLine[2] == "active") // activity status of said md-device
+		personality = mainLine[3]             // Personality type of md-device.
 
 		if len(lines) <= i+3 {
```

Comment:
This check is for the lines per entry though. So far all entries needed to have 3 lines, with raid0 it's only 2 (plus a trailing empty line in both cases). Relying on the fact there should always be another line coming after a raid0 block feels a bit brittle.